### PR TITLE
7260cx3 DualToR config.bcm support based on DualToR setting in device metadata at boot time

### PR DESCRIPTION
**Pull request by gechiang**
#### Why I did it
In order to support DualToR IPinIP functionality for 7260CX3 SKUs there are new BRCM SOC properties required.  We do not want these new SOC properties to be applied for all 7260CX3 SKUs but to base on the DUT role to decide which DUT should include these new SOC properties at boot time.
 
#### How I did it
Added code to check the device metadata to inspect if the DUT is configured for DualToR functionality. Renamed/Updated the current config.bcm file of the supported 7260cx3 SKUs to optionally include the IPinIP required SOC properties based on the DUT DualToR role.  This file is used for "sonic-cfggen" at boot time to dynamically generate the required config.bcm file to be used by BRCM SDK.  This new method eliminated the need to have multiple versions of config.bcm files where the delta is small.

#### How to verify it
Manually changed the device metadata to either include or don't include the DualToR role and verify that the correct bcm.config is installed in the ASIC. 
Here is the verification output for the case where the 7260cx3 is not configured for DualToR purpose:
```
admin@str2-7260cx3-acs-9:~$ docker exec -it syncd bash
root@str2-7260cx3-acs-9:/# cd /etc/sai.d/
root@str2-7260cx3-acs-9:/etc/sai.d# ls -l
total 40
-rw-r--r-- 1 root root 36437 Mar 26 05:20 sai.config
-rw-r--r-- 1 root root    67 Mar 26 05:20 sai.profile
root@str2-7260cx3-acs-9:/etc/sai.d# cat sai.profile
SAI_INIT_CONFIG_FILE=/etc/sai.d/sai.config
SAI_NUM_ECMP_MEMBERS=64
root@str2-7260cx3-acs-9:/etc/sai.d# cat sai.config

PHY_AN_ALLOW_PLL_CHANGE=1
arl_clean_timeout_usec=15000000
asf_mem_profile=2
bcm_num_cos=10
bcm_stat_flags=1
bcm_stat_jumbo=9236
cdma_timeout_usec=15000000
core_clock_frequency=1700
dma_desc_timeout_usec=15000000
dpp_clock_ratio=2:3
higig2_hdr_mode=1
ipv6_lpm_128b_enable=1
l3_alpm_enable=2
lpm_scaling_enable=0
l2xmsg_mode=1
max_vp_lags=0
mem_scan_enable=1
miim_intr_enable=0
module_64ports=1
os=unix
oversubscribe_mode=1
pbmp_xport_xe=0xfffffffd3fffffff4fffffffc7ffffffe
#pbmp_xport_xe=0x47fffffff1fffffffcfffffffe7ffffffe
phy_an_allow_pll_change_hg=0
phy_an_c73=1
phy_chain_rx_lane_map_physical{1.0}=0x3210
phy_chain_rx_lane_map_physical{101.0}=0x0123
...
```
Here is the validation for the case where a 7260CX3 do have DualToR configure:
```
admin@str2-7260cx3-acs-9:~$ bcmcmd "config show" | grep sai_
    sai_tunnel_support=1
admin@str2-7260cx3-acs-9:~$ bcmcmd "config show" | grep ecmp
    l3_ecmp_levels=2
admin@str2-7260cx3-acs-9:~$ bcmcmd "config show" | grep host_as_r
    host_as_route_disable=1

admin@str2-7260cx3-acs-9:~$ docker exec -it syncd bash
root@str2-7260cx3-acs-9:/# cd /etc/sai.d/
root@str2-7260cx3-acs-9:/etc/sai.d# ls -l
total 40
-rw-r--r-- 1 root root 36534 Mar 26 05:32 sai.config
-rw-r--r-- 1 root root    67 Mar 26 05:32 sai.profile
root@str2-7260cx3-acs-9:/etc/sai.d# cat sai.profile
SAI_INIT_CONFIG_FILE=/etc/sai.d/sai.config
SAI_NUM_ECMP_MEMBERS=64
root@str2-7260cx3-acs-9:/etc/sai.d# cat sai.config
            sai_tunnel_support=1
            host_as_route_disable=1
            l3_ecmp_levels=2
PHY_AN_ALLOW_PLL_CHANGE=1
arl_clean_timeout_usec=15000000
asf_mem_profile=2
bcm_num_cos=10
bcm_stat_flags=1
bcm_stat_jumbo=9236
cdma_timeout_usec=15000000
core_clock_frequency=1700
dma_desc_timeout_usec=15000000
dpp_clock_ratio=2:3
higig2_hdr_mode=1
ipv6_lpm_128b_enable=1
l3_alpm_enable=2
lpm_scaling_enable=0
l2xmsg_mode=1
max_vp_lags=0
mem_scan_enable=1
miim_intr_enable=0
module_64ports=1
os=unix
oversubscribe_mode=1
pbmp_xport_xe=0xfffffffd3fffffff4fffffffc7ffffffe
#pbmp_xport_xe=0x47fffffff1fffffffcfffffffe7ffffffe
phy_an_allow_pll_change_hg=0
phy_an_c73=1
phy_chain_rx_lane_map_physical{1.0}=0x3210
phy_chain_rx_lane_map_physical{101.0}=0x0123
phy_chain_rx_lane_map_physical{105.0}=0x0123
phy_chain_rx_lane_map_physical{109.0}=0x0123
...
```

Here is the result when the same code is running on a non 7260cx3 SKU where the config .bcm setup remains unchanged and continue to load from the original config.bcm file without going through "sonic-cfggen":
```
admin@str-s6000-acs-8:~$ docker exec -it syncd bash
root@str-s6000-acs-8:/# cd /etc/sai.d/
root@str-s6000-acs-8:/etc/sai.d# ls -l
total 4
-rw-r--r-- 1 root root 96 Mar 26 05:22 sai.profile
root@str-s6000-acs-8:/etc/sai.d# cat sai.profile
SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/td2-s6000-32x40G.config.bcm
SAI_NUM_ECMP_MEMBERS=32
root@str-s6000-acs-8:/etc/sai.d#
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

#### Description for the changelog
7260cx3 dynamic generate config.bcm based on DualToR role at boot time to add IPinIP support.

### CODE REVIEW

File: platform/broadcom/docker-syncd-brcm/start.sh [hunk 1]
```diff
@@ -8,11 +8,22 @@ SYNCD_SOCKET_FILE=/var/run/sswsyncd/sswsyncd.socket
...
+# If delta is large, use sai.profile.j2 which basically require the user to select which config file to use
+# If delta is small, use append.config.j2 where additional SAI INIT config properties are added
+#   based on specific device metadata requirement
+#   in this case sai.profile should have been modified to use the path /etc/sai.d/sai.config
 
 # Create/Copy the sai.profile to /etc/sai.d/sai.profile
 if [ -f $HWSKU_DIR/sai.profile.j2 ]; then
     sonic-cfggen -d -t $HWSKU_DIR/sai.profile.j2 > /etc/sai.d/sai.profile
 else
+    if [ -f $HWSKU_DIR/append.config.j2 ]; then
```
Comment (prsunny) [inline platform/broadcom/docker-syncd-brcm/start.sh:24]: Should this be done irrespective of  `sai.profile.j2 `exists?

Comment (gechiang) [reply platform/broadcom/docker-syncd-brcm/start.sh:24]: @prsunny These 2 methods are mutual exclusive.  user must select one of the method (using sai.profile.j2 or  append.config.j2) to manage the config.bcm selection.  
Method 1 via sai.profile.j2 is the existing way and this method requires user to have multiple config.bcm files in its entirely to be selected and used.  no modification on top of this due to the file selected is mounted as Read Only.
Method 2 via append.config.j2 allows the user to use one config.bcm file and add additional logic to selectively include some additional SOC properties within the same config file.  But since this method requires dynamic modification, it must be done at the /etc/sai.d directory which requires upecifying this new path in the final sai.profile file (also under /etc/sai.d/).
So in short. this is purposely done so that it will reduce the complexity and user must select which method to specify the config.bcm file.

Comment (prsunny) [reply platform/broadcom/docker-syncd-brcm/start.sh:24]: got it.. thanks @gechiang

**Comment by gechiang** [comment]
@ due to merge conflict introduced by https://github.com/Azure/sonic-buildimage/pull/7027
I am redoing this PR again to absorb the changes from 7027.
Once ready will raise a new PR...
